### PR TITLE
ISPN-9802 externalizing dependencies for productization

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.jboss.org/browse/ISPN-9802

Needs backport into 9.4.x.

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/dependencies/hibernate_v5.2.adoc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[source,xml,options="nowrap"]
+----
+<dependency>
+   <groupId>org.infinispan</groupId>
+   <artifactId>infinispan-hibernate-cache</artifactId>
+   <version>${version.infinispan}</version> <!-- 9.2.x.Final -->
```
Comment (karesti) [inline documentation/src/main/asciidoc/topics/dependencies/hibernate_v5.2.adoc]: is this still true @galderz ? infinispan version

File: documentation/src/main/asciidoc/topics/dependencies/spring_session_dependencies.adoc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            <version>${version.infinispan}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-context</artifactId>
+            <version>${version.spring}</version>
+        </dependency>
+        <dependency>
+           <groupId>org.springframework</groupId>
+           <artifactId>spring-session-core</artifactId>
```
Comment (karesti) [inline documentation/src/main/asciidoc/topics/dependencies/spring_session_dependencies.adoc]: in 9.4 we will have spring 4 and spring 5 too. I did not add the docs yet for the backport I did in the spring 5 PR. 
Concerning this dependency, the value is spring-session for spring 4 and spring-session-core for spring 5

File: documentation/src/main/asciidoc/topics/dependencies/wfly_endpoints.adoc [hunk 3]
```diff
@@ -0,0 +1,11 @@
+[source,xml,options="nowrap"]
+----
+<subsystem xmlns="urn:infinispan:server:endpoint:9.2">
```
Comment (karesti) [inline documentation/src/main/asciidoc/topics/dependencies/wfly_endpoints.adoc]: question, is it normal we do still use 9.2 and not 9.4 here?

File: documentation/src/main/asciidoc/getting_started/embedded_cache.adoc [hunk 4]
```diff
@@ -10,19 +10,15 @@ The only thing you need to set up {brandname} is add it's dependencies to your p
...
-	<version>${infinispan.version}</version>
-</dependency>
-
-----
+include::topics/dependencies/ispn_embedded.adoc[]
 
 to the `<dependencies>` section of the POM.
+
+ifndef::productized[]
 You'll need to substitute `${infinispan.version}` for the version of {brandname} you wish to use.
```
Comment (wburns) [inline documentation/src/main/asciidoc/getting_started/embedded_cache.adoc]: Won't they still need this? Or I guess we need to wait to see what the string placeholder is in the customized ispn_embedded.adoc?

Comment (oraNod) [reply documentation/src/main/asciidoc/getting_started/embedded_cache.adoc]: yeah, I need to rework this. I've been messing about with it and think we can substitute `${infinispan.version}` with a variable that substitutes with the correct value for each release.

File: documentation/src/main/asciidoc/topics/dependencies/wfly_endpoints.adoc (hunk 3 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/dependencies/wfly_endpoints.adoc]: I think this is just one that hasn't been updated. I've fixed it to be 9.4. we'll have to update in 10 again.

File: documentation/src/main/asciidoc/topics/dependencies/hibernate_v5.2.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/dependencies/hibernate_v5.2.adoc]: @karesti I checked with @rvansa here. hibernate v5.2 is supported with 9.2.x only.

File: documentation/src/main/asciidoc/getting_started/installation.adoc [hunk 5]
```diff
@@ -11,27 +11,32 @@ TIP: If you already have any of these pieces of software, there is no need to in
 === JDK
 Choose your Java runtime, and follow their installation instructions. For example, you could choose one of:
 
-*  link:http://openjdk.java.net/install/[OpenJDK] 
+*  link:http://openjdk.java.net/install/[OpenJDK]
 *  link:http://www.java.com/en/download/manual.jsp[Oracle Java SE]
-*  link:http://download.oracle.com/docs/cd/E15289_01/doc.40/e15065/toc.htm[Oracle JRockit] 
+*  link:http://download.oracle.com/docs/cd/E15289_01/doc.40/e15065/toc.htm[Oracle JRockit]
```
Comment (danberindei) [inline documentation/src/main/asciidoc/getting_started/installation.adoc]: Off-topic, but maybe we should remove references to JRockit from the docs.

Comment (oraNod) [reply documentation/src/main/asciidoc/getting_started/installation.adoc]: Good call. Thanks @danberindei I've removed those references and will update the PR.

**Comment by tristantarrant** [comment]
Merged. Please create a backport and I'll close the issue

**Comment by oraNod** [comment]
thank you @tristantarrant ! here's the backport: https://github.com/infinispan/infinispan/pull/6591

**Pull request merged**